src/app/[lng]/(infos)/profile/[walletAddress]/page.tsx

```diff
@@ -61,14 +67,30 @@ export default async function Page({ params }: { params: Params }) {
   }
 
   const sanitizedAddress = result.data;
-  const { data: campaigns } = await getProfileBannerCampaigns();
-  const { data: questsData } = await getQuestsWithNoCampaignAttached();
+  const [{ data: campaigns }, { data: questsData }, isPageEnabled] =
+    await Promise.all([
+      getProfileBannerCampaigns(),
+      getQuestsWithNoCampaignAttached(),
+      getFeatureFlag(GlobalFeatureFlags.ProfilePage),
+    ]);
+
+  const questsExtended = await fetchQuestOpportunitiesByRewardsIds(
+    questsData.data,
+  );
+
+  if (isPageEnabled) {
+    return (
+      <Suspense fallback={<ProfilePageSkeleton />}>
+        <ProfilePage campaigns={campaigns} quests={questsExtended} />
+      </Suspense>
+    );
+  }
+
   return (
-    <ProfilePage
-      quests={questsData.data}
+    <OldProfilePage
+      quests={questsExtended}
       campaigns={campaigns}
       walletAddress={sanitizedAddress}
-      isPublic
     />
   );
 }
```

Comment:
From what I understand we'll use the same init code every time we try to load the profile page component, could we wrap it, and expose a single `<ProfilePage/>` component so that the callers don't need to know about the ab test and the component internals?

Something like:

```ts
export const ProfilePage: React.FC.... () => {
    // load data

  if (isPageEnabled) {
    return (
      <Suspense fallback={<ProfilePageSkeleton />}>
        <ProfilePage2
          isPublic={true}
          campaigns={campaigns}
          quests={questsExtended}
        />
      </Suspense>
    );
  }

  return (
    <ProfilePage1
      isPublic={false}
      quests={questsExtended}
      campaigns={campaigns}
    />
  );
}
```